blik/src/main/java/com/adyen/checkout/blik/BlikViewNew.kt

```diff
@@ -0,0 +1,128 @@
+import com.adyen.checkout.core.log.Logger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+class BlikViewNew @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) :
+    LinearLayout(
+        context,
+        attrs,
+        defStyleAttr
+    ),
+    ComponentViewNew {
+
+    private val binding: BlikViewBinding = BlikViewBinding.inflate(LayoutInflater.from(context), this)
+
+    private lateinit var localizedContext: Context
+
+    private lateinit var blikDelegate: BlikDelegate
+
+    init {
+        orientation = VERTICAL
+        val padding = resources.getDimension(R.dimen.standard_margin).toInt()
+        setPadding(padding, padding, padding, 0)
+    }
+
+    override fun initView(delegate: ComponentDelegate, coroutineScope: CoroutineScope, localizedContext: Context) {
+        if (delegate !is BlikDelegate) throw IllegalArgumentException("Unsupported delegate type")
+        blikDelegate = delegate
+
+        this.localizedContext = localizedContext
+        initLocalizedStrings(localizedContext)
+
+        observeDelegate(delegate, coroutineScope)
+
+        initBlikCodeInput()
+    }
+
+    private fun initLocalizedStrings(localizedContext: Context) {
+        binding.textInputLayoutBlikCode.setLocalizedHintFromStyle(
+            R.style.AdyenCheckout_Blik_BlikCodeInput,
+            localizedContext
+        )
+        binding.textViewBlikHeader.setLocalizedTextFromStyle(
+            R.style.AdyenCheckout_Blik_BlikHeaderTextView,
+            localizedContext
+        )
+    }
+
+    private fun observeDelegate(delegate: BlikDelegate, coroutineScope: CoroutineScope) {
+        delegate.outputDataFlow
+            .onEach { outputDataChanged(it) }
+            .launchIn(coroutineScope)
+    }
+
+    private fun outputDataChanged(blikOutputData: BlikOutputData?) {
+        blikOutputData ?: return
+        // no ops
+    }
+
+    private fun initBlikCodeInput() {
+        binding.editTextBlikCode.setOnChangeListener {
+            blikDelegate.inputData.blikCode = binding.editTextBlikCode.rawValue
+            notifyInputDataChanged()
+            binding.textInputLayoutBlikCode.error = null
+        }
+
+        binding.editTextBlikCode.onFocusChangeListener = OnFocusChangeListener { _: View?, hasFocus: Boolean ->
+            val outputData = blikDelegate.outputData
+            val blikCodeValidation = outputData?.blikCodeField?.validation
+            if (hasFocus) {
+                binding.textInputLayoutBlikCode.error = null
+            } else if (blikCodeValidation != null && !blikCodeValidation.isValid()) {
+                val errorReasonResId = (blikCodeValidation as Validation.Invalid).reason
+                binding.textInputLayoutBlikCode.error = localizedContext.getString(errorReasonResId)
+            }
+        }
+    }
+
+    override val isConfirmationRequired: Boolean
```

Comment:
This could be `override val isConfirmationRequired: Boolean = true`

Reply:
👍